index.js

```diff
@@ -38,40 +57,31 @@ module.exports = fp(async function (fastify, opts) {
 
   fastify.onClose(function () {
     clearInterval(interval)
+    obs.disconnect()
+    observedEntries = {}
   })
-})
 
-function measurements () {
-  return performance.getEntriesByType('measure').filter(e => {
-    return e.name.indexOf(ROUTES) === 0
-  }).reduce((acc, e) => {
-    const key = e.name.split(':')[1]
-    if (acc[key]) {
-      acc[key].push(e.duration)
-    } else {
-      acc[key] = [e.duration]
-    }
-    return acc
-  }, {})
-}
-
-function stats () {
-  const m = measurements()
+  function measurements () {
+    // observedEntries built in PerformanceObserver
+    return observedEntries
+  }
 
-  return Object.keys(m).reduce((acc, k) => {
-    const s = summary(m[k])
-    acc[k] = {
-      mean: s.mean(),
-      mode: s.mode(),
-      median: s.median(),
-      max: s.max(),
-      min: s.min(),
-      sd: s.sd()
-    }
+  function stats () {
+    const m = measurements()
+    observedEntries = {}
 
-    // always clear our measures after stats()
-    performance.clearMeasures(ROUTES + k)
+    return Object.keys(m).reduce((acc, k) => {
+      const s = summary(m[k])
+      acc[k] = {
+        mean: s.mean(),
+        mode: s.mode(),
+        median: s.median(),
+        max: s.max(),
+        min: s.min(),
+        sd: s.sd()
+      }
 
-    return acc
-  }, {})
-}
+      return acc
+    }, {})
+  }
+})
```

Comment:
Can you set the minimum version of fastify to 2.x here? Check the docs of fastify-plugin

Reply:
This breaks a test, looking into it, but FYI may be a minor delay.

Reply:
Updated, but please check the update to the test.

Reply:
I meant something different, it should on the second argument of fp (in other terms, it should not show up in `stats()`.

Reply:
Of course, I see what I did there now.